### ISSUE TITLE
mysql.js compatibility: promise wrapper mysql.format missing

### ISSUE THREAD

**Issue by jacekkopecky** [sidorares/node-mysql2#739]
The recommended use of the promise wrapper is 
```
const mysql = require('mysql2/promise');
```

In the `mysql` package, you can use `format`, `escape`, and `escapeId` on the main package object ([documentation link](https://www.npmjs.com/package/mysql#escaping-query-values)), these functions are only available on connection and pool missing when using `mysql2/promise`. It's not always natural to have a connection or a pool available when you're formatting the queries.

So I think in the interest of compatibility with _mysql.js_, the following code should work:
```
const mysql = require('mysql2/promise');
mysql.escape('foo');
```

**Comment by sidorares** [comment]
yes, worth adding. Looks like a simple change, would you be able to do it yourself @jacekkopecky ?

**Comment by jacekkopecky** [comment]
Hi, I'd love to, but day job will keep me busy in March. If you ping me in April, you have a chance. 8-)

**Comment by sidorares** [comment]
no pressure @jacekkopecky , if I don't find time for this before you I'll ping you in April

**Comment by antixrist** [comment]
closed by [#799](https://github.com/sidorares/node-mysql2/pull/799)
